tests/system.rs

```diff
@@ -263,20 +266,24 @@ fn read_config(filename: &Path) -> Config {
     config
 }
 
-fn format_file<P: Into<PathBuf>>(filepath: P, config: &Config) -> (FileMap, FormatReport) {
+fn format_file<P: Into<PathBuf>>(filepath: P, config: &Config) -> (Summary, FileMap, FormatReport) {
     let filepath = filepath.into();
     let input = Input::File(filepath);
-    let (_error_summary, file_map, report) =
-        format_input::<io::Stdout>(input, config, None).unwrap();
-    (file_map, report)
+    format_input::<io::Stdout>(input, config, None).unwrap()
+}
+
+pub enum IdempotentCheckError {
+    Mismatch(HashMap<PathBuf, Vec<Mismatch>>),
+    Parse,
 }
 
-pub fn idempotent_check(
-    filename: PathBuf,
-) -> Result<FormatReport, HashMap<PathBuf, Vec<Mismatch>>> {
+pub fn idempotent_check(filename: PathBuf) -> Result<FormatReport, IdempotentCheckError> {
     let sig_comments = read_significant_comments(&filename);
     let config = read_config(&filename);
-    let (file_map, format_report) = format_file(filename, &config);
+    let (error_summary, file_map, format_report) = format_file(filename, &config);
+    if error_summary.has_parsing_errors() {
+        return Err(IdempotentCheckError::Parse);
```

Comment:
Changing this to the following resulted in a syntax error.
```rust
Err(IdempotentCheckError::Parse)
```
I don't understand why.

Reply:
Semi-colon-less value as a shorthand for return only works for the last expression in a function

Reply:
😄  Well, that's straightforward!